### PR TITLE
Fix `any`/`all` for Kleene logic

### PR DESCRIPTION
**Pull request by stinodego**
#### Changes

* `any` and `all` in the `boolean_kleene` module now return an `Option<bool>`. _(The versions in the `boolean` module are untouched.)_
* Added doctests for both.

### CODE REVIEW

File: src/compute/boolean_kleene.rs [hunk 1]
```diff
@@ -234,26 +234,70 @@ pub fn and_scalar(array: &BooleanArray, scalar: &BooleanScalar) -> BooleanArray
...
+/// assert_eq!(all(&b), Some(false));
+/// assert_eq!(all(&c), None);
+/// ```
+pub fn all(array: &BooleanArray) -> Option<bool> {
     if array.is_empty() {
-        true
-    } else if array.null_count() > 0 {
-        false
+        Some(true)
+    } else if array.validity().is_some() {
```
Comment (ritchie46) [inline src/compute/boolean_kleene.rs:293]: Can you guard this on `null_count == 0`? There are sometimes cases where we have a validity even though there are no nulls. (due to slicing).

File: src/compute/boolean_kleene.rs [hunk 2]
```diff
@@ -234,26 +234,70 @@ pub fn and_scalar(array: &BooleanArray, scalar: &BooleanScalar) -> BooleanArray
...
+///
+/// assert_eq!(any(&a), Some(true));
+/// assert_eq!(any(&b), Some(false));
+/// assert_eq!(any(&c), None);
+/// ```
+pub fn any(array: &BooleanArray) -> Option<bool> {
     if array.is_empty() {
-        false
+        Some(false)
     } else if array.validity().is_some() {
```
Comment (ritchie46) [inline src/compute/boolean_kleene.rs:259]: See comment below.

File: src/compute/boolean_kleene.rs (hunk 1 above)
Comment (stinodego) [reply src/compute/boolean_kleene.rs:293]: Sure!

Comment (stinodego) [reply src/compute/boolean_kleene.rs]: Done, also updated the same check in the regular boolean `any`.

**Pull request merged**